### PR TITLE
Removed objectManager from ProductHelper

### PR DESCRIPTION
**Pull request by peterjaap**
Also removed useless options array from ImageHelper

**Summary**

Using ObjectManager directly is bad mmmmkay

**Result**
No objectmanager used directly :tada:

### CODE REVIEW

File: Helper/Entity/ProductHelper.php [hunk 1]
```diff
@@ -20,19 +21,22 @@
 use Magento\Catalog\Model\Product\Type\AbstractType;
 use Magento\Catalog\Model\Product\Visibility;
 use Magento\Catalog\Model\ResourceModel\Eav\Attribute as AttributeResource;
+use Magento\Catalog\Model\ResourceModel\Product\Collection;
```
Comment (fooman) [inline Helper/Entity/ProductHelper.php]: I think moving this into the constructor could introduce a subtle change. Previously `objectManager->create()` would provide a new Collection object every time it's run. Moving it into the constructor would make it a shared object. I don't have any Algolia instance to test but from looking at the code using `CollectionFactory` and then `->create()` on it seems safer.

Comment (peterjaap) [reply Helper/Entity/ProductHelper.php]: Thanks, you're right! I changed it to use the factory.

File: Helper/Entity/ProductHelper.php [hunk 2]
```diff
@@ -201,10 +218,8 @@ public function getProductCollectionQuery(
         $onlyVisible = true,
         $includeNotVisibleIndividually = false
     ) {
-        /** @var \Magento\Catalog\Model\ResourceModel\Product\Collection $products */
-        $products = $this->objectManager->create('Magento\Catalog\Model\ResourceModel\Product\Collection');
-
-        $products = $products
+        $productCollection = $this->productCollectionFactory->create();
+        $products = $productCollection
```
Comment (damcou) [inline Helper/Entity/ProductHelper.php]:
```suggestion
        $products = $productCollection
```

Comment (damcou) [reply Helper/Entity/ProductHelper.php]: Hello @peterjaap !
Thanks for this PR and sorry that we didn't have time to handle it before.
Is it possible to submit this change so the codestyle check can pass ?
Thanks in advance ;)

File: Helper/Entity/ProductHelper.php [hunk 3]
```diff
@@ -20,19 +21,22 @@
...
 use Magento\Framework\DataObject;
 use Magento\Framework\Event\ManagerInterface;
-use Magento\Framework\ObjectManagerInterface;
 use Magento\Store\Model\Store;
 use Magento\Store\Model\StoreManagerInterface;
 
 class ProductHelper
 {
+    public $productCollectionFactory;
+    public $groupCollection;
```
Comment (damcou) [inline Helper/Entity/ProductHelper.php]: Could you please make these 2 properties private ?

File: Helper/Image.php [hunk 4]
```diff
@@ -10,8 +10,11 @@
 
 class Image extends \Magento\Catalog\Helper\Image
 {
+    /**
+     * @var ConfigHelper
+     */
+    public $configHelper;
```
Comment (damcou) [inline Helper/Image.php]: Same thing here :) 
Thanks in advance.

File: Helper/Entity/ProductHelper.php (hunk 3 above)
Comment (peterjaap) [reply Helper/Entity/ProductHelper.php]: I'd rather not. Shall we compromise and make them protected? Private leaves me with very few nice options to override these.

File: Helper/Image.php (hunk 4 above)
Comment (peterjaap) [reply Helper/Image.php]: I'd rather not. Shall we compromise and make them protected? Private leaves me with very few nice options to override these.

File: Helper/Entity/ProductHelper.php (hunk 3 above)
Comment (peterjaap) [reply Helper/Entity/ProductHelper.php]: (Right now we have a composer patch in place that makes all private properties and functions in your extension public so we can pluginize/override them - this isn't preferred)

File: Helper/Image.php (hunk 4 above)
Comment (damcou) [reply Helper/Image.php]: Let's go for protected :)

File: Helper/Entity/ProductHelper.php [hunk 5]
```diff
@@ -86,6 +89,25 @@ class ProductHelper
         'color',
     ];
 
+    /**
+     * ProductHelper constructor.
```
Comment (damcou) [inline Helper/Entity/ProductHelper.php]: Hi @peterjaap !
Looks that there's a separation line missing between the description of the constructor and the first param (it breaks the quality-tools check).
Could you please add it and then I think we're good to merge ;)

File: Helper/Entity/ProductHelper.php [hunk 6]
```diff
@@ -86,6 +89,26 @@ class ProductHelper
...
+     *
```
Comment (damcou) [inline Helper/Entity/ProductHelper.php]:
```suggestion
     *
```
@peterjaap Super sorry to ask you to change this again, but the tool is pretty strict :(

**Pull request merged**